Modules/@babylonjs/react-native/EngineHook.ts

```diff
@@ -66,7 +66,48 @@ export function useEngine(): Engine | undefined {
                     }
                 }
 
-                setEngine(engine = new NativeEngine());
+                engine = new NativeEngine();
+
+                // NOTE: This is a workaround for https://github.com/BabylonJS/BabylonReactNative/issues/60
+                let heartbeat: NodeJS.Timeout | null;
+                const startHeartbeat = () => {
+                    if (!heartbeat) {
+                        heartbeat = setInterval(() => {}, 10);
+                    }
+                };
+                const stopHeartbeat = () => {
+                    if (heartbeat) {
+                        clearInterval(heartbeat);
+                        heartbeat = null;
+                    }
+                }
+                startHeartbeat();
+
+                let renderLoopCount = 0;
+
+                const originalRunRenderLoop: (...args: any[]) => void = engine.runRenderLoop;
+                engine.runRenderLoop = function(...args: any[]): void {
+                    originalRunRenderLoop.apply(this, args);
+                    if (++renderLoopCount == 1) {
+                        stopHeartbeat();
+                    }
+                }
+
+                const originalStopRenderLoop: (...args: any[]) => void = engine.stopRenderLoop;
+                engine.stopRenderLoop = function(...args: any[]): void {
+                    if (--renderLoopCount == 0) {
+                        startHeartbeat();
+                    }
+                    originalStopRenderLoop.apply(this, args);
+                }
```

Comment:
So just to make sure I understand this correctly.  We stop our heartbeat task whenever the renderloop starts running, then start it back up whenever the loop goes to sleep just to keep the the looper active?

Reply:
That's right, because when the render loop is running, it is constantly calling `requestAnimationFrame` which goes into native code, and the animation frame callback is invoked through the arcana task scheduler, which explicitly wakes up the run loop. So that ensures the looper (work queue) gets drained, though even still as mentioned above this is probably causing all promise-based async operations to be "throttled." So we almost certainly need a better fix than this, but this should at least unblock usage.